Sources/TripKit/helpers/TKUserProfileHelper.swift

```diff
@@ -132,5 +133,40 @@ public class TKUserProfileHelper: NSObject {
     }
   }
   
+  // MARK: - Mode by mode, Mode picker
+  
+  /// update picked modes provided with all reference modes to determine disabled modes
+  @objc public class func update(pickedModes: Set<TKModeInfo>, allModes: Set<TKModeInfo>) {
+    var modes = disabledSharedVehicleModes
+    
+    let pickedIds = pickedModes.compactMap { try? JSONEncoder().encode($0)  }
+    let allIds = allModes.compactMap { try? JSONEncoder().encode($0)  }
+    
+    let toDisable = allIds.filter {
+      !pickedIds.contains($0)
+    }
+    
+    pickedIds.forEach { mode in
+      if modes.contains(mode) {
+        modes = modes.filter { mode != $0 }
+      }
+    }
```

Comment:
I find this little snippet hard to read. What this does is remove from `modes` everything that's in `pickedIds`, right? 

That's what `mutating func subtract(_ other: [Set])` does. So this can be just:

```suggestion
    modes.subtract(pickedIds)
```

Reply:
that is correct. thats a handy function.